### PR TITLE
Playing State Machine pull request

### PR DESCRIPTION
**Pull request by xKoem**
Created PlayingStateMachine microservice with tests.

### CODE REVIEW

**Comment by boczkas** [comment]
Build 135 Failure

File: PlayingStateMachine/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/PlayerActionEvent.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+import org.springframework.context.ApplicationEvent;
+import pl.jakubowskiprzemyslaw.tajgertim.models.playeraction.PlayerAction;
+
+import java.util.Objects;
+
+public class PlayerActionEvent extends ApplicationEvent {
+
+    private static final long serialVersionUID = 6950694925484781931L;
+    private PlayerAction action;
```
Comment (Hoymm) [inline PlayingStateMachine/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/PlayerActionEvent.java]: final

File: PlayingStateMachine/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/PlayerActionEventListener.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    @Autowired
+    public PlayerActionEventListener(QueueService queueService) {
+        this.queueService = queueService;
+    }
+
+    @Override
+    public void onApplicationEvent(PlayerActionEvent event) {
+        PlayerAction playerAction = event.getPlayerAction();
+
+        if(playerAction.getAction() instanceof Shot) {
```
Comment (Hoymm) [inline PlayingStateMachine/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/PlayerActionEventListener.java]: would be good to do it objectively

File: PlayingStateMachine/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/playingstatemachine/PSMEventListenerTest.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import static org.mockito.Mockito.verify;
+
+@Test
+public class PSMEventListenerTest {
+
+    private PlayerActionEventListener eventListener;
+    private QueueService queueService;
+    private Player player;
+
+    private static final int COORDINATE_X = 1;
```
Comment (totine) [inline PlayingStateMachine/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/playingstatemachine/PSMEventListenerTest.java]: maybe not the same values in X and Y

File: PlayingStateMachine/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/PlayerActionEvent.java (hunk 1 above)
Comment (boczkas) [reply PlayingStateMachine/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/PlayerActionEvent.java]: done

File: PlayingStateMachine/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/PlayerActionEventListener.java (hunk 2 above)
Comment (boczkas) [reply PlayingStateMachine/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/PlayerActionEventListener.java]: will consider it later

[the remark this answers is not in the document]
Comment (boczkas) [reply PlayingStateMachine/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/PlayerActionEventListener.java]: done

File: PlayingStateMachine/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/PlayerActionEventListener.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            queueService.sendObjectToQueue(Queues._10ShotHandlerPlayerShotQueue, playerAction);
+        }
+
+        if(playerAction.getAction() instanceof Move) {
```
[the remark this answers is not in the document]
Comment (boczkas) [reply PlayingStateMachine/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/PlayerActionEventListener.java]: done

File: PlayingStateMachine/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/playingstatemachine/PSMEventListenerTest.java (hunk 3 above)
Comment (boczkas) [reply PlayingStateMachine/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/playingstatemachine/PSMEventListenerTest.java]: done

**Pull request merged**